cfgov/research_reports/models.py

```diff
@@ -0,0 +1,235 @@
+                'children': [{
+                    'title': section3.header,
+                    'body': section3.body,
+                    'url': format('#section-{}.{}.{}', i, j, k),
+                    'numbering': '' if is_appendix else
+                                 format('{}.{}.{} ', i, j, k),
+                } for k, section3 in enumerate(
+                    subsection.report_sections_level_three.all().order_by('pk')
+                )]
+            } for j, subsection in enumerate(
+                section.report_subsections.all().order_by('pk')
+            )]
+        } for i, section in enumerate(sections)]
+
+    return sec
+
+
+get_report_sections = get_report_parts()
+get_report_appendices = get_report_parts(True)
+
+
+def get_researchers():
+    return dict([
+        (r.title, r.url) for r in
+        SublandingFilterablePage.objects.get(pk=4833).get_children()
+    ])
+
+
+def _get_deploy_environment():
+    return getattr(settings, 'DEPLOY_ENVIRONMENT', 'local')
+
+
+class ReportSection(ClusterableModel):
+    header = models.CharField(max_length=200, blank=True)
+    is_appendix = models.BooleanField(default=False)
+    body = models.TextField(blank=True)
+    panels = [
+        FieldPanel('header'),
+        FieldPanel('is_appendix'),
+        FieldPanel('body'),
+        InlinePanel('report_subsections', label='Subsection'),
+    ]
+    report = ParentalKey('ResearchReportPage',
+                         on_delete=models.CASCADE,
+                         related_name='report_sections')
+
+
+class ReportSubSection(ClusterableModel):
+    header = models.CharField(max_length=200)
+    body = models.TextField(blank=True)
+    panels = [
+        FieldPanel('header'),
+        FieldPanel('body'),
+        InlinePanel(
+            'report_sections_level_three',
+            label='Section Level Three'
+        ),
+    ]
+    section = ParentalKey('ReportSection',
+                          on_delete=models.CASCADE,
+                          related_name='report_subsections')
+
+
+class ReportSectionLevelThree(models.Model):
+    header = models.CharField(max_length=200)
+    body = models.TextField(blank=True)
+    subsection = ParentalKey('ReportSubSection',
+                          on_delete=models.CASCADE,
+                          related_name='report_sections_level_three')
+
+
+class ReportAuthor(models.Model):
+    name = models.CharField(max_length=50)
+    report = ParentalKey('ResearchReportPage',
+                         on_delete=models.CASCADE,
+                         related_name='report_authors')
+
+
+class ReportMetaclass(WagtailAdminModelFormMetaclass):
+    @classmethod
+    def child_form(cls):
+        return ReportForm
+
+
+class ReportForm(WagtailAdminPageForm, metaclass=ReportMetaclass):
+    pass
+
+
+class ResearchReportPage(CFGOVPage):
+    # Report Upload Fields
+    report_file = models.ForeignKey(
+        'wagtaildocs.Document',
```

Comment:
If the original reports aren't intended to be downloadable / linked elsewhere on the website, is there an advantage to using a Wagtail Document for this as opposed to a plain Django [`FileField`](https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.FileField)? The latter would still allow for uploading to S3, but would allow for uploading to a different location, and wouldn't stick the uploaded reports among the other Wagtail document views.

Reply:
Yeah, we've discussed this several times before, and I hadn't gotten to that item of my backlog. I had thought I wouldn't tackle that issue yet, but now I realize that it might affect UX, so I probably should. Looking into it now. 👀